pkg/telemetrylogs/condition_builder.go

```diff
@@ -66,9 +66,13 @@ func (c *conditionBuilder) conditionFor(
 		case qbtypes.FilterOperatorNotLike:
 			return sb.NotILike(tblFieldName, value), nil
 		case qbtypes.FilterOperatorRegexp:
-			return fmt.Sprintf(`match(LOWER(%s), LOWER(%s))`, tblFieldName, sb.Var(value)), nil
+			// Note: Escape $$ to $$$$ to avoid sqlbuilder interpreting materialized $ signs
+			// Only needed because we are using sprintf instead of sb.Match (not implemented in sqlbuilder)
+			return fmt.Sprintf(`match(LOWER(%s), LOWER(%s))`, sqlbuilder.Escape(tblFieldName), sb.Var(value)), nil
 		case qbtypes.FilterOperatorNotRegexp:
-			return fmt.Sprintf(`NOT match(LOWER(%s), LOWER(%s))`, tblFieldName, sb.Var(value)), nil
+			// Note: Escape $$ to $$$$ to avoid sqlbuilder interpreting materialized $ signs
+			// Only needed because we are using sprintf instead of sb.Match (not implemented in sqlbuilder)
+			return fmt.Sprintf(`NOT match(LOWER(%s), LOWER(%s))`, sqlbuilder.Escape(tblFieldName), sb.Var(value)), nil
```

Comment:
nit: `body` is already a top level col and it will never have this problem because `tblFieldName` would just be `body` itself.